arches/app/src/components/ControlledListManager/EditLabel.vue

```diff
@@ -0,0 +1,167 @@
+<script setup lang="ts">
+import arches from "arches";
+import { computed, ref } from "vue";
+import { useGettext } from "vue3-gettext";
+
+import Button from "primevue/button";
+import Dialog from "primevue/dialog";
+import Dropdown from "primevue/dropdown";
+import InputText from "primevue/inputtext";
+import { useToast } from "primevue/usetoast";
+
+import { upsertLabel } from "@/components/ControlledListManager/api.ts";
+
+import type {
+    ControlledListItem,
+    Label,
+    LanguageMap,
+} from "@/types/ControlledListManager.d";
+
+const props: {
+    item: ControlledListItem;
+    header: string;
+    languageMap: LanguageMap;
+    label: Label;
+    // updates don't need an insert callback
+    onInsert: null | ((label: Label) => Promise<Label>);
+} = defineProps(["item", "header", "languageMap", "label", "onInsert"]);
+
+const value = ref(props.label.value);
+const language = ref(props.label.language);
+
+const visible = defineModel<boolean>({ required: true });
+
+const languageOptions = computed(() => {
+    if (!props.languageMap) {
+        return [];
+    }
+    return (
+        Object.entries(props.languageMap).map(([code, label]) => {
+            return {
+                label,
+                code,
+            };
+        })
+    );
+});
+
+const toast = useToast();
+const { $gettext } = useGettext();
+const staticItemLabel = $gettext("Item Label");
+const staticLanguageLabel = $gettext("Language");
+
+const onSave = async () => {
+    const upsertedLabel = await upsertLabel(
+        {
+            id: props.label.id,
+            value: value.value,
+            language: language.value,
+            valuetype: props.label.valuetype,
+            itemId: props.item.id,
+        },
+        toast,
+        $gettext,
+    );
+
+    if (upsertedLabel) {
+        if (props.onInsert !== null) {
+            props.onInsert(upsertedLabel);
+            value.value = "";
+        } else {
+            /* eslint-disable vue/no-mutating-props */
+            props.label.language = language.value;
+            props.label.value = value.value;
```

Comment:
mutating props?

Reply:
Props mutation is gone in latest in favor of provide / inject.

Also it was pointed out to me last week at the dev meeting that vuex is more of a vue2 concept; `pinia` is what's new.